### PR TITLE
Adding MRTK page, under GUI

### PR DESCRIPTION
**Pull request by rickfromwork**
Adding a page dedicated to MRTK controls/components. Some of the content doesn't necessarily belong under GUI, but most of it does, and it should all be grouped in the same page.

Open to review/feedback.

### CODE REVIEW

File: content/How_To/GUI/MRTK.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    - title: Babylon 3D GUI
+      url: /divingDeeper/gui/gui3d
+video-overview:
+video-content:
+---
+
+## What is MRTK
+MRTK stands for the Mixed Reality Toolkit, a set of features, controls, and components designed to ease and accelerate development for VR and AR applications.
+
+There currently only exists [documentation for MRTK's Unity implementation](https://docs.microsoft.com/en-us/windows/mixed-reality/mrtk-unity/), and so when we talk about matching what MRTK offers, we are comparing to what this documentation offers.
```
Comment (typride) [inline content/How_To/GUI/MRTK.md]: There is documentation for MRTK for Unreal too:
https://docs.microsoft.com/en-us/windows/mixed-reality/develop/unreal/unreal-mrtk-introduction

File: content/How_To/GUI/MRTK.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+## GUI Controls
+The following BabylonJS controls have roots as MRTK features. As they extend  the `Control3D` class, they all support the basic capabilities that `Control3D` objects have.
+
+More on the `Control3D` class, and GUI elements in general, can be found [here](/divingDeeper/gui/gui).
+
+The following is an example that showcases several of the controls listed, including `TouchHolographicButtons` , `TouchMeshButtons`, and the `NearMenu`. This scene is designed for use in XR, and supports use of `WebXRNearInteraction`:
+<Playground id="#24DLJ4#7" title="Near Interaction Button Scene Demo" description="Demo showcasing different Control3D objects with Near Interaction support."/>
+
+#### Holographic Button
+The `HolographicButton` class is the standard, familiar button that can be placed and pressed in a 3D environment. It is based off of the original MRTK Holographic button, and supports a title, image, and tooltip.
```
Comment (typride) [inline content/How_To/GUI/MRTK.md]: I can't seem to find the holographic button in MRTK 2.6 or 2.7. Wondering if you're referencing the PressableButton features in the documentation? https://docs.microsoft.com/en-us/windows/mixed-reality/mrtk-unity/features/ux-building-blocks/button?view=mrtkunity-2021-05

File: content/How_To/GUI/MRTK.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+See it in action here:
+<Playground id="#2PKNJB#7" title="Slider Color Changer" description="A basic example showing how 3D sliders can affect their surroundings."/>
+
+## Behaviors and Gizmos
+Some behaviors and Gizmos are also shared with MRTK components. In MRTK terminology, these are usually referred to as Solvers, Bounds Control, or Object Manipulators. Behaviors and Gizmos are attached to `Nodes`, `TransformNodes`, and `Meshes` in order to provide additional functionality.
+
+More information on `Gizmos` can be found [here](divingDeeper/mesh/gizmo), while behaviors are covered in more detail [here](divingDeeper/behaviors/meshBehaviors).
+
+#### SixDof Drag Behavior
+The `SixDofDragBehavior`, also known as MRTK's `Object Manipulator` component. By creating and attaching it to an object in the scene, users are able to grab, move, and rotate the object. Interacting with a ray-cast input allows the object to be moved in 3-dimansions, while interacting with near interaction adds the ability to rotate the object in-place.
```
Comment (typride) [inline content/How_To/GUI/MRTK.md]: Spelling: 3-dimensions.

Sentence edit suggestion:
Interacting with a ray-cast input allows an object to be moved in 3-dim**e**nsions, while interacting with near interaction allows for rotation of an object around its center.

File: content/How_To/GUI/MRTK.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+var sixDofDragBehavior = new BABYLON.SixDofDragBehavior();
+```
+To attach a `SixDofDragBehavior` to a mesh in the scene, simply call the mesh's `addBehavior` function:
+```javascript
+mesh.addBehavior(sixDofDragBehavior);
+```
+#### Follow Behavior
+The `FollowBehavior` tries to maintain a set distance between the `TransformNode` it's attached to, and the `followedCamera` target (defaults to main camera).
+
+There are many aspects to this behavior that can be customized, these are the core properties:
```
Comment (typride) [inline content/How_To/GUI/MRTK.md]: Suggestion:
There are many aspects to this behavior that can be customized. These are the core properties:

**Review by typride — COMMENTED** [review]
I've provided some feedback and suggestions throughout. Otherwise, it all looks good to me.

File: content/How_To/GUI/MRTK.md (hunk 2 above)
Comment (rickfromwork) [reply content/How_To/GUI/MRTK.md]: Oh, yes, my bad. I forgot the naming was different.

File: content/How_To/GUI/MRTK.md (hunk 3 above)
Comment (rickfromwork) [reply content/How_To/GUI/MRTK.md]: The wording of that suggestion to me implies that near interaction doesn't support movement in 3 dimensions, only rotation, which would be incorrect. @typride Is that restriction present in existing MRTK implementations?

Comment (typride) [reply content/How_To/GUI/MRTK.md]: Ah, I miswrote. I meant to say "adds the ability to rotate around it's center." 

I was mainly confused with the "in-place" wording.. I think it's good as is.

File: vercel.json [hunk 5]
```diff
@@ -310,6 +310,8 @@
         { "source": "/how_to/gui", "destination": "/divingDeeper/gui/gui" },
         { "source": "/How_To/Gui3D", "destination": "/divingDeeper/gui/gui3D" },
         { "source": "/how_to/gui3d", "destination": "/divingDeeper/gui/gui3D" },
+        { "source": "/How_To/MRTK", "destination": "/divingDeeper/gui/mrtk" },
```
Comment (RaananW) [inline vercel.json]: This file is only meant to keep back-compat linking correct. Was there ever an MRTK page that needs to be linked accordingly? If not you can remove this change.

**Review by RaananW — CHANGES_REQUESTED** [review]
LGTM!

I think more examples (or a line or two of code) would be great for each of the components. For example - show how to enable the eye tracking feature, or link to a different article. Or show a bit of code in the Holographic/Touch holographic button section.

File: content/How_To/GUI/MRTK.md [hunk 6]
```diff
@@ -0,0 +1,147 @@
+---
+title: MRTK 2.X Content
```
Comment (RaananW) [inline content/How_To/GUI/MRTK.md]:
```suggestion
title: MRTK 2.X For Babylon.js
```

File: content/How_To/GUI/MRTK.md [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+      url: /divingDeeper/behaviors/meshBehaviors
+    - title: How To Use Babylon XR Features
+      url: /divingDeeper/webXR/WebXRSelectedFeatures
+    - title: How To Use Babylon AR Features
+      url: /divingDeeper/webXR/webXRARFeatures
+video-overview:
+video-content:
+---
+
+## What is MRTK
```
Comment (RaananW) [inline content/How_To/GUI/MRTK.md]:
```suggestion
## What is MRTK for Babylon.js
```

File: vercel.json [hunk 8]
```diff
@@ -310,6 +310,8 @@
         { "source": "/how_to/gui", "destination": "/divingDeeper/gui/gui" },
         { "source": "/How_To/Gui3D", "destination": "/divingDeeper/gui/gui3D" },
         { "source": "/how_to/gui3d", "destination": "/divingDeeper/gui/gui3D" },
+        { "source": "/How_To/MRTK", "destination": "/divingDeeper/gui/mrtk" },
+        { "source": "/how_to/mrtk", "destination": "/divingDeeper/gui/mrtk" },
```
Comment (RaananW) [inline vercel.json]:
```suggestion
```

**Review by RaananW — APPROVED** [review]

**Comment by RaananW** [comment]
I am merging this, keeping the comments and suggestions in place. We can discuss improvements at a future time.

**Pull request merged**